### PR TITLE
Feature/iia 943 pattern UI flow

### PR DESCRIPTION
**Pull request by dholubek**

### CODE REVIEW

File: kview/src/main/resources/dev/ikm/komet/kview/mvvm/view/pattern/confirmation-pane.fxml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        <HBox alignment="CENTER_RIGHT" BorderPane.alignment="CENTER_RIGHT">
+            <children>
+                <Button fx:id="closePropertiesPanel" alignment="CENTER_RIGHT" contentDisplay="RIGHT" mnemonicParsing="false" onAction="#closeProperties" styleClass="lidr-cancel-clear-button" text="CLOSE PROPERTIES PANEL" textAlignment="CENTER" HBox.hgrow="ALWAYS" />
+                <Region prefHeight="54.0" prefWidth="12.0" />
+            </children>
+        </HBox>
+    </bottom>
+    <styleClass>
+        <String fx:value="lidr-device" />
+        <String fx:value="analyte-group-container" />
```
Comment (carldea) [inline kview/src/main/resources/dev/ikm/komet/kview/mvvm/view/pattern/confirmation-pane.fxml:56]: Should this have lidr information?

Comment (dholubek) [reply kview/src/main/resources/dev/ikm/komet/kview/mvvm/view/pattern/confirmation-pane.fxml:56]: they are just css styles

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/journal/JournalController.java [hunk 2]
```diff
@@ -1391,21 +1442,25 @@ private void makeNewPatternWindow(ObservableViewNoOverride windowView, NidTextEn
...
                 .setPropertyValue(MODULES_PROPERTY, stampViewModel.findAllModules(viewProperties), true);
 
+        StateMachine patternSM = StateMachine.create(new PatternDetailsPattern());
+
+        ValidationViewModel patternViewModel = new PatternViewModel()
+                .setPropertyValue(VIEW_PROPERTIES, viewProperties)
+                .setPropertyValue(MODE, CREATE)
+                .setPropertyValue(STAMP_VIEW_MODEL, stampViewModel)
+                .setPropertyValue(PATTERN_TOPIC, UUID.randomUUID())
+                .setPropertyValue(STATE_MACHINE, patternSM);
```
Comment (carldea) [inline kview/src/main/java/dev/ikm/komet/kview/mvvm/view/journal/JournalController.java:1452]: **Suggestion:**
In another ticket, we could improve how we are using the state machine we can do the following:

- Don't pass the state machine in the ViewModel. Leave it inside of the `PropertiesController`. Events and subscribers can listen and make transitions. Any state from the `PatternDetailsController` such as property toggle open or closed. 
- Use the state machine's `.when()` methods. Here you can perform actions when a state is encountered. https://github.com/carldea/axonic/wiki#when-code-block-executions

**Review by carldea — APPROVED** [review]
Look good. As long as it works as designed via figma and user workflow.

I added notes and suggestions to improve the use of the state machine code.

**Pull request merged**